UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/HealthCheck/AzureWebJobsHealthCheckClient.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.Common.Configuration;
+using UKHO.ExchangeSetService.Common.Helpers;
+using UKHO.ExchangeSetService.Common.Models.Enums;
+
+namespace UKHO.ExchangeSetService.Common.HealthCheck
+{
+    [ExcludeFromCodeCoverage]
+    public class AzureWebJobsHealthCheckClient : IAzureWebJobsHealthCheckClient
+    {
+        static HttpClient httpClient = new HttpClient();
+        private readonly IOptions<EssFulfilmentStorageConfiguration> essFulfilmentStorageConfiguration;
+        private readonly IWebJobsAccessKeyProvider webJobsAccessKeyProvider;
+        private readonly IWebHostEnvironment webHostEnvironment;
+        private readonly IAzureBlobStorageService azureBlobStorageService;
+
+        public AzureWebJobsHealthCheckClient(IOptions<EssFulfilmentStorageConfiguration> essFulfilmentStorageConfiguration,
+                                             IWebJobsAccessKeyProvider webJobsAccessKeyProvider,
+                                             IWebHostEnvironment webHostEnvironment,
+                                             IAzureBlobStorageService azureBlobStorageService)
+        {
+            this.essFulfilmentStorageConfiguration = essFulfilmentStorageConfiguration;
+            this.webJobsAccessKeyProvider = webJobsAccessKeyProvider;
+            this.webHostEnvironment = webHostEnvironment;
+            this.azureBlobStorageService = azureBlobStorageService;
+        }
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                string webJobUri, userNameKey, passwordKey = string.Empty;
+                string[] exchangeSetTypes = essFulfilmentStorageConfiguration.Value.ExchangeSetTypes.Split(",");
+                List<Tuple<string, string, string, int>> webJobs = new List<Tuple<string, string, string, int>>();
```

Comment:
In this instance I wonder whether a List<WebJobDetails> would offer better clarity?

Reply:
Fixed, As suggested have made required changes.